src/Microsoft.VisualStudio.Services.Agent/Logging.cs

```diff
@@ -1,32 +1,52 @@
 using Microsoft.VisualStudio.Services.Agent.Util;
 using System;
-using System.Globalization;
-using System.Threading;
+using System.IO;
 
 namespace Microsoft.VisualStudio.Services.Agent
 {
-    public interface ILogWriter
-    {
-        Guid TimeLineId { get; set; }
-        void Write(string message);
-    }
-
     [ServiceLocator(Default = typeof(PagingLogger))]
-    public interface IPagingLogger : IAgentService, ILogWriter
+    public interface IPagingLogger : IAgentService
     {
+        void Setup(Guid timelineId, Guid timelineRecordId);
+
+        void Write(string message);        
+
+        void End();
     }
 
     public class PagingLogger : AgentService, IPagingLogger
     {
-        IJobServer _jobServer;
+        public const int PageSize = 256;
+
+        private Guid _timelineId;
+        private Guid _timelineRecordId;
+        private string _pageId;
+        private FileStream _pageData;
+        private StreamWriter _pageWriter;
+        private int _lineCount;
+        private int _pageCount;
+        private string _dataFileName;
+        private string _pagesFolder;
+        private IJobServerQueue _jobServerQueue;
 
         public override void Initialize(IHostContext hostContext)
         {
             base.Initialize(hostContext);
-            _jobServer = hostContext.GetService<IJobServer>();
+            _pageWriter = null;
+            _pageData = null;
+            _lineCount = 0;
+            _pageCount = 0;
+            _pageId = Guid.NewGuid().ToString();
+            _pagesFolder = Path.Combine(IOUtil.GetDiagPath(), "pages");
```

Comment:
not sure whether we want to put all page log into a new folder or just put them in diag folder.